packages/graphiql-toolkit/src/create-fetcher/lib.ts

```diff
@@ -92,10 +92,58 @@ export const createWebsocketsFetcherFromUrl = (
  */
 export const createWebsocketsFetcherFromClient = (wsClient: Client) => (
   graphQLParams: FetcherParams,
-) =>
-  makeAsyncIterableIteratorFromSink<FetcherResult>(sink =>
-    wsClient!.subscribe(graphQLParams, sink),
-  );
+): AsyncIterableIterator<FetcherResult> => {
+  let deferred: {
+    resolve: (done: boolean) => void;
+    reject: (err: unknown) => void;
+  } | null = null;
+  const pending: FetcherResult[] = [];
+  let throwMe: unknown = null,
+    done = false;
+  const dispose = wsClient.subscribe<FetcherResult>(graphQLParams, {
+    next: data => {
+      pending.push(data);
+      deferred?.resolve(false);
+    },
+    error: err => {
+      if (err instanceof Error) {
+        throwMe = err;
+      } else if (err instanceof CloseEvent) {
+        throwMe = new Error(
+          `Socket closed with event ${err.code} ${err.reason || ''}`.trim(),
+        );
+      } else {
+        throwMe = new Error(
+          (err as GraphQLError[]).map(({ message }) => message).join(', '),
+        );
+      }
+      deferred?.reject(throwMe);
+    },
+    complete: () => {
+      done = true;
+      deferred?.resolve(true);
+    },
+  });
+  return {
+    [Symbol.asyncIterator]() {
```

Comment:
Note that Symbol.asyncIterator is not available in Safari 14

Reply:
Also, shouldn't the transpilation be handled by the user? If you'd like to support Safari 14 on iOS, simply polyfill the async iterator with Babel on build step. @acao what is your take on this?

Reply:
I have no strong feelings about this. For myself, I can say that I am shipping an app with Safari 14 support and try to avoid babel transforms where possible, that is why I created the lib. If GraphiQL does not need to rely on push-pull-async-iterable-iterator I am totally fine with it.

Reply:
Reverted back to using the push-pull-async-iterable-iterator following v2.1.3.